### ISSUE TITLE
Deleting post crashes

### ISSUE THREAD

**Issue by yuriiz** [ellmetha/django-machina#72]
Deleting post from /forum/.../topic/.../.../post/delete/ crashes with following traceback:
```

File "/app/.heroku/python/lib/python2.7/site-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/app/.heroku/python/lib/python2.7/site-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)

File "/app/.heroku/python/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/app/.heroku/python/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/app/.heroku/python/lib/python2.7/site-packages/newrelic/hooks/framework_django.py" in wrapper
  504.                 return wrapped(*args, **kwargs)

File "/app/.heroku/python/lib/python2.7/site-packages/django/views/generic/base.py" in view
  68.             return self.dispatch(request, *args, **kwargs)

File "/app/.heroku/python/lib/python2.7/site-packages/machina/apps/forum_permission/viewmixins.py" in dispatch
  108.         return super(PermissionRequiredMixin, self).dispatch(request, *args, **kwargs)

File "/app/.heroku/python/lib/python2.7/site-packages/newrelic/hooks/framework_django.py" in wrapper
  886.             return wrapped(*args, **kwargs)

File "/app/.heroku/python/lib/python2.7/site-packages/django/views/generic/base.py" in dispatch
  88.         return handler(request, *args, **kwargs)

File "/app/.heroku/python/lib/python2.7/site-packages/django/views/generic/edit.py" in post
  269.         return self.delete(request, *args, **kwargs)

File "/app/.heroku/python/lib/python2.7/site-packages/django/views/generic/edit.py" in delete
  264.         self.object.delete()

File "/app/.heroku/python/lib/python2.7/site-packages/machina/apps/forum_conversation/abstract_models.py" in delete
  320.             self.topic.delete()

File "/app/.heroku/python/lib/python2.7/site-packages/machina/apps/forum_conversation/abstract_models.py" in delete
  178.         super(AbstractTopic, self).delete(using)

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/base.py" in delete
  972.         return collector.delete()

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/deletion.py" in delete
  308.                             sender=model, instance=obj, using=self.using

File "/app/.heroku/python/lib/python2.7/site-packages/django/dispatch/dispatcher.py" in send
  193.             for receiver in self._live_receivers(sender)

File "/app/.heroku/python/lib/python2.7/site-packages/machina/apps/forum_member/receivers.py" in decrease_posts_count
  76.     profile.save()

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/base.py" in save
  806.                        force_update=force_update, update_fields=update_fields)

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/base.py" in save_base
  836.             updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/base.py" in _save_table
  903.                                       forced_update)

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/base.py" in _do_update
  953.         return filtered._update(values) > 0

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/query.py" in _update
  664.         return query.get_compiler(self.db).execute_sql(CURSOR)

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/sql/compiler.py" in execute_sql
  1191.         cursor = super(SQLUpdateCompiler, self).execute_sql(result_type)

File "/app/.heroku/python/lib/python2.7/site-packages/django/db/models/sql/compiler.py" in execute_sql
  886.             raise original_exception

Exception Type: IntegrityError at /groups/forum/glc-14/topic/bride-trafficking-11/15/post/delete/
Exception Value: new row for relation "forum_member_forumprofile" violates check constraint "forum_member_forumprofile_posts_count_check"
DETAIL:  Failing row contains (7, , null, -1, null, 703).
```

I suppose reason is that `posts_count` must be positive and `decrease_posts_count` can try to write negative value in it.

**Comment by ellmetha** [comment]
@yuriiz are you able to reproduce this error from an empty forum? I noticed you wrote a patch to make the `decrease_posts_count` more robust; would you mind creating a PR so that it get merged into the project?

**Comment by yuriiz** [comment]
No, I can't reproduce it from empty forum, I do not know who did it get to this state. Pull request created: https://github.com/ellmetha/django-machina/pull/75

**Comment by ellmetha** [comment]
Fixed in 2a1d45f.
